CMP1903M A01 2223/Pack.cs

```diff
@@ -42,27 +42,28 @@ public static bool shuffleCardPack(int typeOfShuffle)
             if (typeOfShuffle == 1) // This checks if the type of shuffles value is one, if it is it will continue with the function
             {
                 var random = new Random(); // This sets the value of the random variable to a random number
-                for (int Num_Of_Cards = pack.Count; Num_Of_Cards > 1; Num_Of_Cards--)
+                for (int Num_Of_Cards = pack.Count ; Num_Of_Cards > 1; Num_Of_Cards--) // a for loop which as long as the number of cards is bigger than one will loop through the following
                 {
-                    int Card_To_Swap = random.Next(Num_Of_Cards);
+                    int Card_To_Swap = random.Next(Num_Of_Cards); // This defines card to swap as a random integer smaller than or equal to the number of cards in the pack
 
-                    Card temp = pack[Card_To_Swap];
-                    pack[Card_To_Swap] = pack[Num_Of_Cards - 1];
-                    pack[Num_Of_Cards - 1] = temp;
+                    Card temp = pack[Card_To_Swap]; // creates a temporary file to hold the random card that was selected
+                    pack[Card_To_Swap] = pack[Num_Of_Cards - 1]; //The card that has been randomly selected is removed from the pack so there are no duplicates
+                    pack[Num_Of_Cards - 1] = temp; // copies the temporary file into the pack
                 }
                 return true;
             }
             //Riffle Shuffle
-            else if (typeOfShuffle == 2) // if the value is 2 it will proceed with this function
+            else if (typeOfShuffle == 2) // if the value passed through the function is 2 it will proceed with this function
             {
-                int Pack_Pointer = 0;
-                while (Pack_Pointer != pack.Count / 2)
+                int Pack_Pointer = 0; // Sets the value of the pack pointer to zero
+                while (Pack_Pointer != pack.Count / 2) // While the pack pointer does not equal to the size of the pack divided by two  this loop will run
```

Comment:
To make it easier to understand the code and make it easier to read i would rename Pack_pointer to something small like j or just pointer, having such a long name is unnecessary in this case